File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []
    q = deque()
    tmp_stack = []
    for i in range(len(progresses)):
        remain_process = 100 - progresses[i]
        days = (remain_process // speeds[i]) + (0 if remain_process % speeds[i] == 0 else 1)
        q.append(days)
    
    start = q[0]
    tmp = []
    cnt = 0
    
    while q:
        print("q",q)
        print("tmp",tmp)
        print("start",start)
        target = q.popleft()
        if target <= start:
            tmp.append(target)
        else:
            answer.append(len(tmp))
            start = target
            tmp = [target]
           
    if len(tmp)>0:
        answer.append(len(tmp))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> answerList = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        
        for (int i=0; i< progresses.length; i++){
            int remain_pg = 100-progresses[i];
            
            int need_time = remain_pg / speeds[i];
            
            if (remain_pg % speeds[i] !=0){
                need_time +=1;
            }
            
            q.add(need_time);
        
        }
        
//         int[] answer = new int[q.size()];
//         for (int i=0; i<q.size(); i++){
//             answer[i] = q.get(i);
//         }
        
        int cnt = 1;
        int pivot = q.poll(); //popleft와 같은..
        
        while (!q.isEmpty()){
            int now = q.poll();
            
            if (now <= pivot){
                cnt+=1;
            }
            else{
                answerList.add(cnt);
                cnt = 1;
                pivot = now;
            }
        }
        answerList.add(cnt);
        int [] answer = new int[answerList.size()];
        for(int i=0 ; i<answerList.size(); i++){
            answer[i] = answerList.get(i); //arrayList에서 값을 뽑아오려면, answerList[i]하면 에러가 남. 꼭 get으로 뽑아와야함
        }
        return answer;
    }
}